modulemd/v2/modulemd-component.c

```diff
@@ -226,6 +226,45 @@ modulemd_component_set_property (GObject *object,
 }
 
 
+static gboolean
+modulemd_component_default_equals (ModulemdComponent *self, ModulemdComponent *other) 
+{
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT (self), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_COMPONENT (other), FALSE);
+
+  if (self == other) return TRUE;
+
+  gint64 buildorder_self = modulemd_component_get_buildorder (self);
+  gint64 buildorder_other = modulemd_component_get_buildorder (other);
+
+  if (buildorder_self != buildorder_other) return FALSE;
```

Comment:
Don't bother creating new variables for these comparisons; just compare the function results directly. You only need to create a variable if the function returns a newly-allocated object or pointer (so you can free it). When getting a numerical or `const char *` back, you can just use it directly.